plugins/woocommerce/src/Internal/FraudProtection/SessionClearanceManager.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * SessionClearanceManager class file.
+ */
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\FraudProtection;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Manages session clearance state for fraud protection.
+ *
+ * This class handles the session status tracking for fraud protection decisions,
+ * managing three possible states: pending, allowed, and blocked. It integrates
+ * with WooCommerce sessions and uses the FraudProtectionController logging helper
+ * to maintain consistent audit logs.
+ *
+ * @since 10.5.0
+ * @internal This class is part of the internal API and is subject to change without notice.
+ */
+class SessionClearanceManager {
+
+	/**
+	 * Session key for storing clearance status.
+	 */
+	private const SESSION_KEY = '_fraud_protection_clearance_status';
+
+	/**
+	 * Session status: pending clearance.
+	 */
+	public const STATUS_PENDING = 'pending';
+
+	/**
+	 * Session status: allowed.
+	 */
+	public const STATUS_ALLOWED = 'allowed';
```

Comment:
Would it be worth having a challenge cleared status to differentiate between low-risk and cleared sessions?